src/Microsoft.NetCore.Analyzers/Core/Security/ReviewCodeForRegexInjectionVulnerabilities.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.NetCore.Analyzers.Security.Helpers;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public class ReviewCodeForRegexInjectionVulnerabilities : SourceTriggeredTaintedDataAnalyzerBase
+    {
+        internal static readonly DiagnosticDescriptor Rule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA3012",
```

Comment:
Wondering if you want to move the IDs to a common type to avoid clashes, unless this is being caught be our analyzer that flags duplicate IDs used within the same project by different diagnostic analyzers.

Reply:
Good idea.  I've just been searching the code manually.

Reply:
Will do something in a future PR.